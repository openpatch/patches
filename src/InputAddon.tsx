import { Box } from "./Box";

export type InputAddonProps = {
  children?: string;
  className?: string;
};

export const InputAddon = ({ children, ...props }: InputAddonProps) => {
  return (
    <Box
      as="span"
      minWidth="auto"
      paddingY="xxsmall"
      paddingX="xsmall"
      textColor="neutral.400"
      backgroundColor="neutral.50"
      {...props}
    >
      {children}
    </Box>
  );
};
