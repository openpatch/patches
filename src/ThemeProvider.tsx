import React, { ReactElement } from "react";
import { ThemeProvider as EThemeProvider } from "emotion-theming";
import { Theme, base } from "./themes";
import { GlobalCssTheme } from "./themes/utils";

interface ThemeProviderProps {
  theme?: Theme;
  children: ReactElement;
}

const ThemeProvider = ({ theme = base, children }: ThemeProviderProps) => {
  return (
    <EThemeProvider theme={theme}>
      <GlobalCssTheme theme={theme} />
      {children}
    </EThemeProvider>
  );
};

export default ThemeProvider;
