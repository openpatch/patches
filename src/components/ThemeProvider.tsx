import React, { ReactElement } from "react";
import {
  ThemeProvider as SCThemeProvider,
  DefaultTheme,
} from "styled-components";
import defaultTheme from "../shared/theme";

interface ThemeProviderProps {
  theme?: DefaultTheme;
  children: ReactElement;
}

const ThemeProvider = ({
  theme = defaultTheme,
  children,
}: ThemeProviderProps) => {
  return <SCThemeProvider theme={theme}>{children}</SCThemeProvider>;
};

export default ThemeProvider;
