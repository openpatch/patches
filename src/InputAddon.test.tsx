
import { render } from "@testing-library/react";
import { InputAddon, InputAddonProps } from "./InputAddon";
import { ThemeProvider } from "./ThemeProvider";

const renderWithTheme = (props?: InputAddonProps) =>
render(
    <ThemeProvider>
        <InputAddon {...props} />
    </ThemeProvider>
);

test("should render", () => {
   renderWithTheme();
});

test("should match snapshot", () => {
    const { container } = renderWithTheme();
    expect(container).toMatchSnapshot();
});