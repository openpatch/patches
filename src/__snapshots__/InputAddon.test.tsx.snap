// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  min-width: auto;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
  color: #7B8794;
  background-color: #F5F7FA;
}

<div>
  <div
    id="patches-root"
  >
    <span
      class="emotion-0"
    />
  </div>
</div>
`;
